File: 1605. Find Valid Matrix Given Row and Column Sums/restorematrix.go
package main

import . "leetcode/utils"

// 贪心算法
func restoreMatric(rowSum, colSum []int) [][]int {
	row, col := len(rowSum), len(colSum)
	matrix := make([][]int, row)
	for x := range matrix {
		matrix[x] = make([]int, col)
	}
	for i := range rowSum {
		for j := range colSum {
			m := Min(rowSum[i], colSum[j]) // 最大可填值
			matrix[i][j] = m
			rowSum[i] -= m // 更新和值
			colSum[j] -= m
		}
	}
	return matrix
}

func main() {
	var (
		rowSum = []int{5, 7, 10}
		colSum = []int{8, 6, 8}
	)
	restoreMatric(rowSum, colSum)
}

File: 1605. Find Valid Matrix Given Row and Column Sums/restorematrix.js
(function (rowSum, colSum) {
    const row = rowSum.length, col = colSum.length
    const matrix = new Array(row).fill(0).map(() => new Array(col).fill(0))
    let i = 0, j = 0
    while (i < row && j < col) {
        const v = Math.min(rowSum[i], colSum[j])
        matrix[i][j] = v
        rowSum[i] -= v
        colSum[j] -= v
        if (rowSum[i] === 0) {
            ++i
        }
        if (colSum[j] === 0) {
            ++j
        }
    }
    return matrix
})()